clang-tools-extra/clang-tidy/modernize/RawStringLiteralCheck.cpp

```diff
@@ -58,7 +58,7 @@ bool containsEscapedCharacters(const MatchFinder::MatchResult &Result,
       *Result.SourceManager, Result.Context->getLangOpts());
   StringRef Text = Lexer::getSourceText(CharRange, *Result.SourceManager,
                                         Result.Context->getLangOpts());
-  if (Text.empty() || isRawStringLiteral(Text))
+  if (Text.empty() || !Text.contains('"') || isRawStringLiteral(Text))
```

Comment:
Why is this necessary?

Reply:
It picked up on this:

```cpp
#define TRICK(arg_) #arg_
char const *const MacroBody = TRICK(foo\\bar);
```

Since `foo\\bar` does not have quotes, the assertion inside `isRawStringLiteral` failed.

Reply:
So... shouldn't the check include something that says this is only when the text is a macro argument to `#`, or am I missing something?

Reply:
I'm not sure I understand the question. Do you mean that we should update the user-facing documentation of the check? Or that we should modify the implementation?

Reply:
I mean this is a blanket-wide check to see if the text contains a double quote character (`"`), but isn't contextualized to the specific case that matters -- when the text is supplied as an macro argument to the preprocessor string-ize operator `#`.  I don't see how adding this condition applies _only_ to that situation.

Reply:
Yeah, that won't work. Do note this PR is abandoned :)

Reply:
Yes, I just wanted to understand the change as written to make sure I wasn't missing something `:)`.  FYI, I am the one who contributed this check and a weakness of several early checks I wrote is that they don't play well with macros and this was one of them.